ContentsInventory, ContentsRoom 제작

hun2ya9 opened the pull request:
#세부사항
1. Inventory
 - 원하는 Contents 유형 선택하면 해당 Contents 들을 불러와서 UI상에 띄움
 - Contents 선택하면 정보를 받아올 수 있음
 - Disposer과 연동 필요

2. ContentsRoom
 - 모든 Contents를 방에 적절히 스폰하여 시각적인 효과를 극대화 시킴
 - Contents를 선택한 경우 Mouse Input 쪽과 연동을 하여 처리해야 함
 - 단순히 일정 간격으로 배치한 것이기 때문에 아직 디자인 적인 부분에서는 미흡한 점이 많음

3. AssetDataLoad
 - Object Type으로 루트에서 부터 데이터를 긁어오는 함수를 임시로 MinusiLibrary 파일에 만들어 두었음
 - 나중에 Load, Save 기능을 하는 모듈을 만든다면 그쪽으로 이전한다거나 해야할 것

hun2ya9 committed:
Loader _ Build.cs 파일에 AssetRegistry 모듈 등록

1. Build.cs 파일에 AssetRegistry 모듈 등록
 - 비동기적으로 로드할 Asset을 검색하고 관리 하는데 효율적인 모듈

2. Loader 클래스
 - 설정된 Object 타입에 해당하는 Asset들을 각각의 배열에 저장함
 - 원하는 Object 타입의 Asset배열을 반환

hun2ya9 committed:
Inventory _ 아이템, 에셋 인벤토리 제작, 로더 제거

1. 아이템 인벤토리
 - 아이템을 담고 꺼낼 수 있는 용도
 - 어떻게 활용될지는 모르겠지만 보통의 게임에 존재하는 그 인벤토리

2. 에셋 인벤토리(뷰어)
 - 에셋을 항목 별로 보여주고 꺼내는 용도
 - 현재 에셋 인벤토리에서 에셋을 누르면 에셋의 AssetData를 받아오는 것이 가능함
 - 에셋 경로와 오브젝트 형과 이름을 조합하여 특정 폴더 안이나 데이터 테이블 등에서 에셋에 알맞은 이미지를 불러올 수 있도록 해야함

3. 로더
 - 에셋을 불러오는 기능이 AssetRegistry에 의해서 아주 간단하게 가능함에 따라 만들어둔 Loader 는 제거하였고, 함수 라이브러리에 임시로 옮겨둠
 -  추후에 로드, 세이브 등을 담당할 모듈에 옮겨질 예정

hun2ya9 committed:
ContentsRoom _ 컨텐츠 스폰, 배치 구현

1. 구현 사항
 - 원점을 기준으로 일정 거리 떨어진 4방향에 오브젝트 유형별로 StaticMeshActor를 스폰하여 배치
 - 원점에서 부터의 거리, 한 행에 생성할 갯수, 좌우 간격, 상하 간격 등을 변수로 설정 간으
 - 너무 큰 Mesh의 경우 Component Bound의 Sphere Raidus값을 일정 크기 이상이면 그 값으로 Scale을 나눔
   간단하게 설명하자면 Mesh가 일정한 크기로 존재하도록 조절

2. 미흡점
 - 원래 맵과 컨텐츠 룸이 있는 맵을 오가는 기능 미구현
 - 아직 선반이나 벽 등은 배치하지 않았음
 - 컨텐츠를 클릭하는 기능은 Mouse Input 쪽과 연동이 필요함

hun2ya9 committed:
Fix _ 폴더 위치 변경에 따른 오류 수정

1. 오류 수정
 - 폴더 위치가 변경되어 Gizmo에 사용하던 Mesh의 material 위치를 찾지 못해서 뜨는 오류 수정함

hun2ya9 committed:
Merge branch 'prototype' into Inventory

Minusi commented on UEPrototype/Source/UEPrototype/Public/Utility/MinusiFrameworkLibrary.h:
```diff
@@ -8,8 +8,25 @@
 #include "Kismet/BlueprintFunctionLibrary.h"
 #include "Kismet/KismetMathLibrary.h"
 #include "Kismet/KismetSystemLibrary.h"
+#include "AssetRegistry/Public/AssetData.h"
```
헤더에서 사용될 필요가 없는 라이브러리는 cpp에 #include 선언하고, 헤더에 있는 포인터형은 전부 전방 선언으로 돌릴것

hun2ya9 replied:
전방선언 완료 불필요한 헤더파일 제거

Minusi commented on UEPrototype/Source/UEPrototype/Public/Inventory/Inventory.h:
```diff
@@ -0,0 +1,39 @@
+﻿// Fill out your copyright notice in the Description page of Project Settings.
```
클래스, 프로퍼티, 함수에 대한 주석 포함할 것

hun2ya9 replied:
불필요한 클래스로 제거됨

Minusi commented on UEPrototype/Source/UEPrototype/Public/Inventory/Item.h:
```diff
@@ -0,0 +1,55 @@
+﻿// Fill out your copyright notice in the Description page of Project Settings.
```
클래스, 프로퍼티, 함수에 대한 주석 포함할 것

hun2ya9 replied:
불필요한 클래스로 제거됨

Minusi commented on UEPrototype/Source/UEPrototype/Public/Inventory/ItemInventory.h:
```diff
@@ -0,0 +1,59 @@
+﻿// Fill out your copyright notice in the Description page of Project Settings.
```
클래스, 프로퍼티, 함수에 대한 주석 포함할 것

hun2ya9 replied:
불필요한 클래스로 삭제

Minusi commented on UEPrototype/Source/UEPrototype/Public/Utility/MinusiFrameworkLibrary.h:
```diff
@@ -8,8 +8,25 @@
 #include "Kismet/BlueprintFunctionLibrary.h"
```
클래스, 프로퍼티, 함수에 대한 주석 포함할 것

hun2ya9 replied:
추가완료

Minusi commented on UEPrototype/Source/UEPrototype/Public/Inventory/AssetInventory.h:
```diff
@@ -0,0 +1,54 @@
+﻿// Fill out your copyright notice in the Description page of Project Settings.
```
클래스, 프로퍼티, 함수에 대한 주석 포함할 것

hun2ya9 replied:
추가완료

hun2ya9 committed:
Inventory _ 불필요한 파일 제거 및 주석 처리, ContentsRoom C++포팅

1. 수정 사항
 - Item 인벤토리와 Item는 사용하지 않을 것으로 예상하여 제거
 - ContentsRoom의 변수와 대부분 함수를 C++로 옮김

hun2ya9 committed:
Merge remote-tracking branch 'origin/Inventory' into Inventory

hun2ya9 committed:
Merge _ 충돌 파일 수정

1. 충돌 파일 수정
 - MinusiFrameworkLibrary를 동시에 수정해서 저장소 것을 가져옴
 - 합병 이후에 내 것을 가져와서 추가할 예정
